docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md

```diff
@@ -0,0 +1,156 @@
+
+For additional information, including parameters and properties, see the [AWS documentation on AWS APIGatewayV2 API](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-api.html).
+
+## Installation
+
+{{% inspec_aws_install %}}
+
+## Syntax
+
+Ensure that the api exists.
+
+```ruby
+describe aws_api_gateway_v2_api(api_id: 'REST_API_ID') do
+  it { should exist }
+end
+```
+
+## Parameters
+
+`api_id` _(required)_
+: The API ID.
+
+## Properties
+
+`api_endpoint`
+: The URI of the API, of the form {api-id}.execute-api.{region}.amazonaws.com.
+
+`api_gateway_managed`
+: Specifies whether an API is managed by API Gateway. You can't update or delete a managed API by using API Gateway. A managed API can be deleted only through the tooling or service that created it.
+
+`api_id`
+: The API ID.
+
+`api_key_selection_expression`
+: An API key selection expression. Supported only for WebSocket APIs.
+
+`cors_configuration`
+: A CORS configuration. Supported only for HTTP APIs.
+
+`cors_configuration.allow_credentials`
+: Specifies whether credentials are included in the CORS request. Supported only for HTTP APIs.
+
+`cors_configuration.allow_headers`
+: Specifies whether credentials are included in the CORS request. Supported only for HTTP APIs.
+
+`cors_configuration.allow_methods`
+: Represents a collection of allowed HTTP methods. Supported only for HTTP APIs.
+
+`cors_configuration.allow_origins`
+: Represents a collection of allowed origins. Supported only for HTTP APIs.
+
+`cors_configuration.expose_headers`
+: Represents a collection of exposed headers. Supported only for HTTP APIs.
+
+`cors_configuration.max_age`
+: The number of seconds that the browser should cache preflight request results. Supported only for HTTP APIs.
+
+`created_date`
+: The timestamp when the API was created.
+
+`description`
+: The description of the API.
+
+`disable_schema_validation`
+: Avoid validating models when creating a deployment. Supported only for WebSocket APIs.
+
+`disable_execute_api_endpoint`
+: Specifies whether clients can invoke your API by using the default execute-api endpoint. By default, clients can invoke your API with the default https://{api_id}.execute-api.{region}.amazonaws.com endpoint. To require that clients use a custom domain name to invoke your API, disable the default endpoint.
+
+`import_info`
+: The validation information during API import.
+
+`name`
+: The name of the API.
+
+`protocol_type`
+: The API protocol.
+
+`route_selection_expression`
+: The route selection expression for the API. For HTTP APIs, the routeSelectionExpression must be ${request.method} ${request.path}. If not provided, this will be the default for HTTP APIs. This property is required for WebSocket APIs.
+
+`tags`
+: A collection of tags associated with the API.
+
+`version`
+: A version identifier for the API.
+
+`warnings`
+: The warning messages reported when `failonwarnings` is turned on during API import.
+
```

Comment:
Also, one global query - dont we have to mention about the property type - string, integer, boolean, etc ?

Reply:
We can add but this change must be done to all the resources which will be a lot of work.

Reply:
Ok, lets once check with Clinto too!